sacc/sacc.py

```diff
@@ -877,34 +952,48 @@ def _add_2pt(self, data_type, tracer1, tracer2, x, tag_val, tag_name, window, tr
             t = {tag_name: float(tag_val)}
             if window is not None:
                 t['window'] = window
-            self.add_data_point(data_type, (tracer1, tracer2), x, tracers_later=tracers_later, **t)
+                t['window_id'] = window_id
```

Comment:
OK, this is an option, but I'm not sure it'll reduce complexity. From now on `get_ell_cl` and `get_theta_xi` will have to return a variable number of arguments (since we need to return the index too), and this number will depend not only on the input arguments (e.g. `return_window`), but also on the type of window the 2-point function is associated with.

I think it's not too far-fetched to associate one window object per two-point function, and I would say this would also be useful for cluster counts (e.g. where each observed richness bin would correspond to a probability distribution of true richness over the same array of true richnesses).

I'm happy to consider something along the lines of what you suggest if we can come up with a way to make these function calls simple to parse, but I think the current 2D `Window` class would be useful beyond just angular power spectra.